### PR TITLE
SLS-399 - set status confirmed when mail gets sent

### PR DESCRIPTION
**Pull request by ebeutler**
https://synjira.atlassian.net/browse/SLS-399

### CODE REVIEW

File: src/main/java/com/celements/course/service/CourseService.java [hunk 1]
```diff
@@ -570,9 +570,20 @@ private boolean isParticipantInState(XWikiDocument regDoc, String emailAdr,
 
   @Override
   public boolean sendConfirmationMail(DocumentReference regDocRef, int participantObjNb) {
-    return XWikiObjectFetcher.on(modelAccess.getOrCreateDocument(regDocRef))
-        .filter(participantObjNb).stream().findFirst()
-        .map(this::sendConfirmationMail).orElse(false);
+    XWikiDocument regDoc = modelAccess.getOrCreateDocument(regDocRef);
+    boolean sendSuccess = XWikiObjectFetcher.on(regDoc).filter(participantObjNb).stream()
+        .findFirst().map(this::sendConfirmationMail).orElse(false);
+    if(sendSuccess)  {
```
Comment (fpichler) [inline src/main/java/com/celements/course/service/CourseService.java]: 1. do not change semantic of sendConfirmationMail. Not all sending of a confirmation Mail may change the ParticipantStatus to confirmed. Only if a confirmation mail is send trough the backend the status may change to confirmed if it is unconfirmed
2. ensure that a status of payed, cancelled and duplicate is not changed. Only the "unconfirmed" may change to confirmed.

**Review by fpichler — CHANGES_REQUESTED** [review]

**Comment by fpichler** [comment]
@msladek wdyt?

File: src/main/java/com/celements/course/service/CourseService.java (hunk 1 above)
Comment (msladek) [reply src/main/java/com/celements/course/service/CourseService.java]: I agree, I wouldn't expect `sendConfirmationMail` to alter the registration doc as a client. I'd rather see an additional explicit call in `sendParticipantConfirmationMail.vm`.

File: src/main/java/com/celements/course/service/CourseService.java [hunk 2]
```diff
@@ -570,9 +570,20 @@ private boolean isParticipantInState(XWikiDocument regDoc, String emailAdr,
...
+      try {
+        XWikiObjectEditor.on(regDoc).filter(participantObjNb)
+            .editField(CourseParticipantClass.FIELD_STATUS).first(ParticipantStatus.confirmed);
+        modelAccess.saveDocument(regDoc, "set confirmed after confirmation mail was sent");
```
Comment (msladek) [inline src/main/java/com/celements/course/service/CourseService.java]:
```suggestion
        boolean changed = XWikiObjectEditor.on(regDoc).filter(participantObjNb)
            .editField(CourseParticipantClass.FIELD_STATUS)
            .first(ParticipantStatus.confirmed);
        if (changed) {
          modelAccess.saveDocument(regDoc, "set confirmed after confirmation mail was sent");
        }
```

File: src/main/java/com/celements/course/service/CourseService.java (hunk 1 above)
Comment (msladek) [inline src/main/java/com/celements/course/service/CourseService.java]:
```suggestion
    if (sendSuccess) {
```

**Review by msladek — CHANGES_REQUESTED** [review]

**Review by fpichler — DISMISSED** [review]

**Comment by ebeutler** [comment]
Fixed Formatting

**Review by msladek — APPROVED** [review]

**Pull request merged**